internal/envoy/auth.go

```diff
@@ -61,45 +62,50 @@ func UpstreamTLSContext(ca []byte, subjectName string, sni string, alpnProtocols
 	// is an interface, returning nil from validationContext directly into
 	// this field boxes the nil into the unexported type of this grpc OneOf field
 	// which causes proto marshaling to explode later on. Not happy Jan.
-	vc := validationContext(ca, subjectName)
+	vc := validationContext(ca, subjectName, true)
 	if vc != nil {
 		context.CommonTlsContext.ValidationContextType = vc
 	}
 
 	return context
 }
 
-func validationContext(ca []byte, subjectName string) *envoy_api_v2_auth.CommonTlsContext_ValidationContext {
+func validationContext(ca []byte, subjectName string, subjectRequired bool) *envoy_api_v2_auth.CommonTlsContext_ValidationContext {
```

Comment:
Do we really need the `subjectRequired` here? Since at the API level, the representation of upstream and downstream policy is disjoint, users should not be able to specify a subject name for downstream policy.

Reply:
I removed it. I needed to remove also the checks `len(ca)` and `len(subjectName)` from `validationContext()` and move these to `UpstreamTLSContext()`. I originally added `subjectRequired` since latter was not valid for `DownstreamTLSContext()`.